keps/sig-api-machinery/20190507-kube-controller-manager-config.md

```diff
@@ -0,0 +1,478 @@
+---
+kep-number: 40
+title: Kube-Controller Manager Config
+authors:
+  - "@luxas"
+  - "@stewart-yu"
+  - "@sttts"
+owning-sig: sig-api-machinery
+participating-sigs:
+  - sig-cluster-lifecycle
+reviewers:
+  - "@deads2k"
+  - "@liggitt"
+  - "@luxas"
+  - "@sttts"
+  - "@thockin"
+approvers:
+  - "@deads2k"
+  - "@liggitt"
+  - "@luxas"
+  - "@sttts"
+  - "@thockin"
+editor:
+  name: "@stewart-yu"
+creation-date: 2019-05-07
+last-updated: 2019-05-22
+status: provisional
+---
+
+# Kube-Controller Manager Config
+
+**How we can start supporting configure kube-controller manager via a config file instead of command line flags.**
+
+## Table of Contents
+
+- [Kube-Controller Manager Config](#kube-controller-manager-config)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [How will the component config file look like](#how-will-the-component-config-file-look-like)
+    - [How is it versioned](#how-is-it-versioned)
+  - [Design Details](#design-details)
+    - [Struct for flags](#struct-for-flags)
+	- [Structs for serializable config](#structs-for-serializable-config)
+	- [Bootstrapping and running the controller manager server](#bootstrapping-and-running-the-controller-manager-server)
+  - [Remaining work](#remaining-work)
+    - [Add typemeta to every sub-controller configuration](#add-typemeta-to-every-sub-controller-configuration)
+    - [Examples](#examples)
+       - [Alpha -> Beta Graduation](#alpha---beta-graduation)
+       - [Beta -> GA Graduation](#beta---ga-graduation)
+       - [Removing a deprecated flag](#removing-a-deprecated-flag)
+    - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+
+## Summary
```

Comment:
Maybe I've still got a bit of jet lag, but I am having a really hard time understanding these first few sections.